src/components/VmStatus/VmStatus.js

```diff
@@ -19,72 +22,87 @@ import {
-    Importing
-    {additionalText}
-  </LinkStatus>
-);
-
-const StateV2VConversionInProgress = ({ progress, ...props }) => (
-  <LinkStatus icon="import" {...props}>
-    Importing (VMware)
-  </LinkStatus>
+const VmStatusInProgress = ({ header, message, children, progress, linkTo, linkMessage }) => (
+  <PopoverStatus icon="in-progress" header={header}>
+    <StatusDescriptionField>{message}</StatusDescriptionField>
+    {children}
+    {progress && <StatusProgressField title={IMPORTING} progress={progress} />}
+    <StatusLinkField title={linkMessage} linkTo={linkTo} />
+  </PopoverStatus>
 );
-StateV2VConversionInProgress.defaultProps = {
+VmStatusInProgress.defaultProps = {
   progress: null,
+  children: null,
 };
-StateV2VConversionInProgress.propTypes = {
+VmStatusInProgress.propTypes = {
+  header: PropTypes.string.isRequired,
+  message: PropTypes.string.isRequired,
+  linkTo: PropTypes.oneOfType([PropTypes.node, PropTypes.func]).isRequired,
+  linkMessage: PropTypes.string.isRequired,
+  children: PropTypes.any,
   progress: PropTypes.number,
 };
 
-const StateV2VConversionPending = ({ ...props }) => (
-  <LinkStatus icon="pending" {...props}>
-    Importing Pending (VMware)
-  </LinkStatus>
-);
-
-const StateV2VConversionError = ({ ...props }) => (
-  <LinkStatus icon="error-circle-o" {...props}>
-    Importing Error (VMware)
-  </LinkStatus>
+const VmStatusPending = ({ header, message, children, linkTo, linkMessage }) => (
+  <PopoverStatus icon="hourglass-half" iconType="fa" header={header}>
+    <StatusDescriptionField>{message}</StatusDescriptionField>
+    {children}
+    <StatusLinkField title={linkMessage} linkTo={linkTo} />
+  </PopoverStatus>
 );
-StateImporting.defaultProps = {
-  additionalText: undefined,
+VmStatusPending.defaultProps = {
+  children: null,
 };
-StateImporting.propTypes = {
-  additionalText: PropTypes.string,
+VmStatusPending.propTypes = {
+  header: PropTypes.string.isRequired,
+  message: PropTypes.string.isRequired,
+  linkTo: PropTypes.oneOfType([PropTypes.node, PropTypes.func]).isRequired,
+  linkMessage: PropTypes.string.isRequired,
+  children: PropTypes.any,
 };
-const StateError = ({ children, ...props }) => (
-  <LinkStatus icon="error-circle-o" {...props}>
+
+const VmStatusError = ({ header, message, children, linkTo, linkMessage }) => (
+  <PopoverStatus icon="error-circle-o" header={header}>
+    <StatusDescriptionField>{message}</StatusDescriptionField>
     {children}
-  </LinkStatus>
+    <StatusLinkField title={linkMessage} linkTo={linkTo} />
+  </PopoverStatus>
 );
-StateError.propTypes = {
-  children: PropTypes.any.isRequired,
+VmStatusError.defaultProps = {
+  children: null,
+};
+VmStatusError.propTypes = {
+  header: PropTypes.string.isRequired,
+  message: PropTypes.string.isRequired,
+  linkMessage: PropTypes.string.isRequired,
+  linkTo: PropTypes.oneOfType([PropTypes.node, PropTypes.func]).isRequired,
+  children: PropTypes.any,
 };
 
 export const VmStatuses = props => {
```

Comment:
We should also move the importerPods statuses into the popover. Then we should either show the same exact status + popover in the vmDetails or show the inline version with just the content there and without the popover part. @andybraren @matthewcarleton thoughts?

not sure if you want to do it in this PR or leave it for the followup...

Reply:
For now let's use the same [Status Popover pattern](http://openshift.github.io/openshift-origin-design/web-console/4.0-designs/status/status.html#status-popovers) within VM Details as well (so no inline version).

FYI, we have some longer-term plans to improve Resource Details pages console-wide using some of our Dashboard card-related work, which would allow us to show error statuses/messages inline. But we'll save that for later. Another popover is fine. 🙂

Reply:
+1 @andybraren